homeassistant/components/tfl/sensor.py

```diff
@@ -0,0 +1,160 @@
+
+from datetime import timedelta
+import logging
+from operator import itemgetter
+from urllib.error import HTTPError, URLError
+
+from tflwrapper import stopPoint
+
+from homeassistant.components.sensor import (  # ENTITY_ID_FORMAT,; PLATFORM_SCHEMA,
+    SensorEntity,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TfLConfigEntry
+from .common import call_tfl_api
+from .const import (
+    ATTR_NEXT_ARRIVALS,
+    ATTR_NEXT_THREE_ARRIVALS,
+    CONF_STOP_POINTS,
+    DOMAIN,
+    RAW_ARRIVAL_DESTINATION_NAME,
+    RAW_ARRIVAL_LINE_NAME,
+    RAW_ARRIVAL_TIME_TO_STATION,
+)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: TfLConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the TfL sensor."""
+    stop_point_api = entry.runtime_data.stop_point_api
+
+    conf = entry.options
+
+    stop_point_ids: list[str] = conf[CONF_STOP_POINTS]
+
+    try:
+        stop_point_infos = await call_tfl_api(
+            hass, stop_point_api.getByIDs, stop_point_ids, False
+        )
+        devices = []
+        if isinstance(stop_point_infos, list):
+            for idx, stop_point_id in enumerate(stop_point_ids):
+                devices.append(
+                    StopPointSensor(
+                        stop_point_api,
+                        stop_point_infos[idx]["commonName"],
+                        stop_point_id,
+                        entry.entry_id,
+                    )
+                )
+        else:
+            devices.append(
+                StopPointSensor(
+                    stop_point_api,
+                    stop_point_infos["commonName"],
+                    stop_point_ids[0],
+                    entry.entry_id,
+                )
+            )
+
+        async_add_entities(devices, True)
+
+    except HTTPError as exception:
+        error_code = exception.code
+        _LOGGER.exception(
+            "Error retrieving stop point data from TfL for stop_points=%s with HTTP error_code=%s, entities will not be created",
+            stop_point_ids,
+            error_code,
+        )
+    except URLError as exception:
+        _LOGGER.exception(
+            "Error retrieving stop point data from TfL for stop_points=%s with URLError reason=%s, entities will not be created",
+            stop_point_ids,
+            exception.reason,
+        )
+
+
+class StopPointSensor(SensorEntity):
+    """Representation of a TfL StopPoint as a Sensor."""
+
+    _attr_attribution = "Powered by TfL Open Data"
+    _attr_icon = "mdi:bus"
+
+    def __init__(
+        self, stop_point_api: stopPoint, name: str, stop_point_id: str, entry_id: str
+    ) -> None:
+        """Initialize the TfL StopPoint sensor."""
+        # super().__init__(coordinator)
```

Comment:
How come the integration doesn't use a coordinator?

Reply:
TfL doesn't have an API that allows you to retrieve the arrivals for multiple stop points at the same time.

[This is the api used](https://zackaryh8.github.io/tfl-api-wrapper-py/#tflwrapper.stopPoint.getStationArrivals), from the `tflwrapper` python library.

My understanding of the use case for the `DataUpdateCoordinator` ([docs](https://developers.home-assistant.io/docs/integration_fetching_data/)) is when you can have a single API poll for all the entities. That's not the case here. There is a 1:1 mapping between an API call and an entity so it fits under the _Separate polling for each individual entity_ use case.

Reply:
OK, I guess that makes sense. I'd suggest to add a (very brief) explanation to the sensor class' docstring explaining this.

Reply:
Done in f9d00ea263b8fcc94079b83ebe4622c0fb244d90